packages/discovery-provider/src/api/v1/users.py

```diff
@@ -2538,11 +2538,12 @@ class FullPurchasesCount(Resource):
         params={"id": "A User ID"},
     )
     @full_ns.expect(purchases_and_sales_count_parser)
-    @auth_middleware(purchases_and_sales_count_parser, require_auth=True)
+    @auth_middleware(purchases_and_sales_count_parser)
     @full_ns.marshal_with(purchases_count_response)
     def get(self, id, authed_user_id):
         decoded_id = decode_with_abort(id, full_ns)
-        check_authorized(decoded_id, authed_user_id)
+        if decoded_id and not is_authorized_request(decoded_id):
+            abort(403, message="You are not authorized to access this resource")
```

Comment:
nit: I don't love repeating this logic everywhere. It feels like a step backwards since we have to remember to duplicate it correctly each time.